rest/rest-api/src/main/java/org/apache/brooklyn/rest/domain/EntityConfigSummary.java

```diff
@@ -45,16 +54,32 @@ public EntityConfigSummary(
             @JsonProperty("label") String label,
             @JsonProperty("priority") Double priority,
             @JsonProperty("possibleValues") List<Map<String, String>> possibleValues,
+            @JsonProperty("pinned") Boolean pinned,
+            @JsonProperty("constraints") List<String> constraints,
             @JsonProperty("links") Map<String, URI> links) {
         super(name, type, description, defaultValue, reconfigurable, label, priority, possibleValues);
+        this.pinned = pinned;
+        this.constraints = (constraints == null) ? ImmutableList.<String>of() : ImmutableList.copyOf(constraints);
         this.links = (links == null) ? ImmutableMap.<String, URI>of() : ImmutableMap.copyOf(links);
     }
 
-    public EntityConfigSummary(ConfigKey<?> config, String label, Double priority, Map<String, URI> links) {
+    public EntityConfigSummary(ConfigKey<?> config, String label, Double priority, Boolean pinned, Map<String, URI> links) {
         super(config, label, priority);
+        this.pinned = pinned;
+        this.constraints = !config.getConstraint().equals(Predicates.alwaysTrue())
```

Comment:
It's unexpected to me but predicates don't override the `equals` method so they follow the default behaviour. Why this comparison works is because the implementation of `alwaysTrue` is an enum. Worth a comment.

What they usually do is override the `toString` method so the most reliable means for comparing predicates is comparing their `toString` values. We are dealing with a well defined predicate here so no need to change.